tests/framework/Optimizers/tests

```diff
@@ -110,38 +87,40 @@
     type = 'RavenFramework'
     input = 'beale_with_sampler.xml'
     csv = 'BealeWithSampler/opt_export.csv'
+    zero_threshold = 1e-2
     rel_err = 1.e-6
   [../]
   [./TrajGrid]
     type = 'RavenFramework'
     input = 'traj_grid.xml'
     csv = 'TrajGrid/opt_export.csv'
-    rel_err = 1.e-6
+    rel_err = 1.e-3
   [../]
   [./BealeFiniteDifference]
     type = 'RavenFramework'
     input = 'beale_finite_difference.xml'
     csv = 'BealeFiniteDifference/opt_export.csv'
-    rel_err = 1.e-6
+    zero_threshold = 1.e-2
   [../]
   [./StochasticFiniteDifference]
     type = 'RavenFramework'
     input = 'stochastic_finite_difference.xml'
     csv = 'StochasticFiniteDifference/opt_export.csv'
-    rel_err = 1.e-6
+    rel_err = 1.e-2
   [../]
   [./BealeFailureRejection]
     type = 'RavenFramework'
     input = 'beale_failure_and_rejection.xml'
     csv = 'BealeFailureRejection/opt_export.csv'
+    zero_threshold = 3.e-1
     rel_err = 1.e-6
   [../]
 
   [./RrR]
     type = 'RavenFramework'
     input = 'raven_running_raven.xml'
     csv = 'RavenRunningRaven/outer.csv'
-    rel_err = 1.e-6
+    rel_err = 1.e-2
```

Comment:
Test on different machines follow somewhat different paths that start out very close together, but small changes quickly become large changes over few iterations. I considered the intention of the test, and the changed tests are not trying to guarantee the path to the solution, but rather that a reasonable solution is reached given the short time available to run a test.

Because different operating systems were following different paths but ending up with reasonably similar results, I don't see a better way to test this functionality. This will also prevent regolding (and a large burden on the reviewer) every time a small change is made that affects only high-precision comparison. There are still tests that carefully check the paths followed (the requirements test) to a higher precision, that will ensure the operation of the system is consistent.

Reply:
As discussed, heavy tests need to be added.

Reply:
Issue #891 added to track this need.